### PR TITLE
quickly exit on error returns

### PR DESCRIPTION
**Pull request by baentsch**
"Guesstimate" for fixing #165 

Is it possible that errors raised but not immediately acted up (stopping any further execution) can trigger follow-on problems in openssl?

### CODE REVIEW

**Comment by baentsch** [comment]
@levitte: May I pick your brain and ask the question above to you: If "ERR_raise" is not followed by stopping execution immediately (as only now done via this PR), could mayhem such as reported in #165 ensue? Why then didn't this manifest earlier & on other platforms? 

The more I think about it, it looks like I really misunderstood `ERR_raise` when writing the original code: 
I was under the assumption that (ID) registration errors are no reasons to terminate initialization, as no problem would occur if those algorithms (not successfully registered) would never be used. And if they were used (and failed), the error queue entry generated (by the original code) would "point the way".

In other words: Is merging this PR essential for correct operation of `oqsprovider`?

**Comment by levitte** [comment]
Your assumption about `ERR_raise` are correct as far as I understand you.  All they do is record an error record into the error queue.  Period.  What happens in #165 is something different, as you can see from my answer there.

**Comment by baentsch** [comment]
Thanks, @levitte for the feedback. So we _could_ leave the logic as-is.

So, asking @dstebila @bhess @mouse07410 for their opinion: Should `oqsprovider` fail quickly when any object registration error is reported by `openssl` (as per this PR) or keep running until `openssl` encounters a "real"/operational error (as is the case with the current "main" branch)? 

I personally now tend to the former, given we have issues where `openssl` crashes before reporting errors, so setting this "Ready for Review".

**Comment by dstebila** [comment]
> Thanks, @levitte for the feedback. So we _could_ leave the logic as-is.
> 
> So, asking @dstebila @bhess @mouse07410 for their opinion: Should `oqsprovider` fail quickly when any object registration error is reported by `openssl` (as per this PR) or keep running until `openssl` encounters a "real"/operational error (as is the case with the current "main" branch)?
> 
> I personally now tend to the former, given we have issues where `openssl` crashes before reporting errors, so setting this "Ready for Review".

I don't know enough about this situation to have an informed opinion.

**Review by dstebila — APPROVED** [review]
